graf2d/gpad/src/TCanvas.cxx

```diff
@@ -779,7 +779,7 @@ void TCanvas::Close(Option_t *option)
 {
    TPad    *padsave = (TPad*)gPad;
    TCanvas *cansave = 0;
-   if (!IsBatch()) if (padsave) cansave = (TCanvas*)gPad->GetCanvas();
+   if (padsave && gPad) cansave = (TCanvas*)gPad->GetCanvas();
```

Comment:
This sounds stupid ...padsave is equal to gPad on line 780... did I missed soothing ?

Reply:
Humm ... right .. I missed that ... this means that the original problem need to be debugged again to find the "real" problem.

Reply:
It seems the fix is:
```
   if (padsave && gPad->TestBit(kNotDeleted)) cansave = (TCanvas*)gPad->GetCanvas();
```
I know this test is not always 100% sure but at least with this test, pad4 example dos not crash and stressGraphics is ok. And it is more logical as it is all about a double delete of the pad. @andriish can you check it solve your problem and make the appropriate PR ?